### DISCUSSION TITLE
Not doing basic validation on unrelated projects?

### DISCUSSION THREAD

**Discussion by JACONDE** [Dynatrace/dynatrace-configuration-as-code#206] [Archive]
Related to #55 
As @UnseenWizzard wrote 
"In general the tool reads in all configs - at which point the env var resolution currently happens - in order to be able to handle dependencies between projects.
That is admittedly an imperfect solution for having several projects, or in-progress projects, because as you noticed they all are validated..."
I think this creates additional complexity because all project is validated and if any environment variable is not applicable for an specific project, you need to create all variables in all projects, otherwise it fails.

**Comment by tatomir146** [comment]
@JACONDE all projects are only read if no specific project is specified (missing -p parameter). Mixing configurations and projects is not allowed.

```
- projects
  - A
  - B
  - C
```
In the above example if you specify `-p="A"` only project A is read, except if there are dependencies defined in Project A to e.g. Projects B and/or C. In that case projects B and C are then also included.

**Comment by JACONDE** [comment]
My project folder looks like this:

projects
> General
> > application
> > app-detection-rule
> Application
> > alerting-profile
> > notification

when executing ./monaco -v -e="./projects/tenants.yaml" -p="./projects/application" projects, it checks also the dependencies also for General project.
As confirmed by @UnseenWizzard , "_In general the tool reads in all configs - at which point the env var resolution currently happens - in order to be able to handle dependencies between projects.
That is admittedly an imperfect solution for having several projects, or in-progress projects, because as you noticed they all are validated..._"
I feel unnecessary, to perform validations on projects with no dependencies, because (for instance) you need to create all the environment variables from all projects in the project you would like to run, otherwise you are getting errors, even do some of those variables are not used at all in the project called.

**Comment by tatomir146** [comment]
We need all projects loaded before we start searching for dependencies. Maybe we could think of changing the way we are looking for dependencies.

**Reply by kristofre** [reply]
@tatomir146 it would make sense to first read the specified projects and validate those and only read dependant projects when they are referenced in the specified projects

**Comment by JACONDE** [comment]
It will makes sense for me if monaco checks the dependencies only for the projects required.
In my case, I'm automating the configuration using Azure DevOps and I'm defining the environment variables by pipeline.
Doing so, I need to add manually all the variables to each pipeline, even do some of them are not used.

**Comment by tatomir146** [comment]
This will be solved with monaco config 2.0
